### ISSUE TITLE
Collect and transport metrics

### ISSUE THREAD

**Issue by Ilheu** [Azure/iotedge#5238]
I'm trying to add the module Azure IoT Edge Metrics Collector in edge, but I'm getting some errors:

1. 

 **I don't understand how to add these URL's** 
![image](https://user-images.githubusercontent.com/8053606/125322514-c7cffa80-e335-11eb-92b7-839ead2d4d13.png)

2.
**I get these errors when I check the logs of the module:**

[2021-07-12 16:21:08.909 INF] Scraping finished, received 0 metrics from endpoint http://edgeHub:9600/metrics
[2021-07-12 16:21:08.907 ERR] Error scraping endpoint http://edgeAgent:9600/metrics - System.Net.Http.HttpRequestException: Name does not resolve
 ---> System.Net.Sockets.SocketException (0xFFFDFFFF): Name does not resolve
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.Devices.Edge.Azure.Monitor.MetricsScraper.ScrapeEndpoint(String endpoint, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-modules/azure-monitor/src/MetricsScraper.cs:line 72
[2021-07-12 16:21:08.910 INF] Scraping finished, received 0 metrics from endpoint http://edgeAgent:9600/metrics
[2021-07-12 16:21:09.819 DBG] 200 OK 
[2021-07-12 16:21:09.821 INF] Successfully sent 0 metrics to fixed set table
[2021-07-12 16:21:09.821 INF] Successfully completed periodic operation Scrape and Upload Metrics

**Comment by veyalla** [comment]
Thanks for the report @Ilheu.

Need a bit more information about your environment. Are attempting to use via a proxy server? Is there a firewall blocking access to Log Analytics endpoints?

Also, please review the monitoring troubleshooting page at: https://docs.microsoft.com/azure/iot-edge/how-to-troubleshoot-monitoring-and-faq?view=iotedge-2020-11&pivots=metrics-collection

**Comment by Ilheu** [comment]
**Need a bit more information about your environment.**
What do you need? I can show everything I have in edge... I'm trying to monitor my edge with this new module.

**Are attempting to use via a proxy server?** 
I'm attempting to use AzureMonitor as UploadTarget. 

**Is there a firewall blocking access to Log Analytics endpoints?**
No.

I've set these env settings: 
![image](https://user-images.githubusercontent.com/8053606/125328439-3748e880-e33c-11eb-9b4f-eb6a9752f807.png)

**Comment by veyalla** [comment]
So I gather no firewall blocking or proxy in use?

The `httpSettings__enabled` should not be `false` on the built-in metrics endpoints which means the edgeAgent and edgeHub containers (not metrics-collector). Also need 1.1.x or 1.2.x edgeAgent and edgeHub versions.

**Comment by Ilheu** [comment]
**So I gather no firewall blocking or proxy in use?** Correct

**The httpSettings__enabled should not be false on the built-in metrics endpoints which means the edgeAgent and edgeHub containers (not metrics-collector).**  I have changed to edgeAgent and edgeHub containers.

![image](https://user-images.githubusercontent.com/8053606/125420470-baed3507-f682-4c91-a482-df69d573df77.png)
![image](https://user-images.githubusercontent.com/8053606/125425067-9b733e96-eb8a-40c3-9eb7-7f6e0540696b.png)



**This error continues:**


muvu@muvu-tech:~$ iotedge logs IoTEdgeMetricsCollector
[2021-07-13 08:15:44 : Starting Module
[2021-07-13 08:15:44.751 INF] Initializing metrics collector
[2021-07-13 08:15:45.125 INF] Version - 1.0.0
[2021-07-13 08:15:45.140 DBG] Waiting on shutdown handler to trigger
[2021-07-13 08:15:45.565 INF] Metrics collector initialized with the following settings:
Settings:
LogAnalyticsWorkspaceId=**************************
Endpoints=[
  "http://edgeHub:9600/metrics",
  "http://edgeAgent:9600/metrics"
]
ScrapeFrequencySecs=300
UploadTarget=AzureMonitor
CompressForUpload=True
TransformForIoTCentral=False
AllowedMetrics=
BlockedMetrics=
ResourceId=********************************************************
[2021-07-13 08:15:46.634 DBG] Starting periodic operation Scrape and Upload Metrics...
[2021-07-13 08:15:46.677 INF] Scraping endpoint http://edgeHub:9600/metrics
[2021-07-13 08:15:46.811 INF] Scraping endpoint http://edgeAgent:9600/metrics
[2021-07-13 08:15:46.828 INF] Started operation Scrape and Upload Metrics
[2021-07-13 08:15:47.019 ERR] Error scraping endpoint http://edgeHub:9600/metrics - System.Net.Http.HttpRequestException: Name does not resolve
 ---> System.Net.Sockets.SocketException (0xFFFDFFFF): Name does not resolve
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.Devices.Edge.Azure.Monitor.MetricsScraper.ScrapeEndpoint(String endpoint, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-modules/azure-monitor/src/MetricsScraper.cs:line 72
[2021-07-13 08:15:47.019 ERR] Error scraping endpoint http://edgeAgent:9600/metrics - System.Net.Http.HttpRequestException: Name does not resolve
 ---> System.Net.Sockets.SocketException (0xFFFDFFFF): Name does not resolve
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.Devices.Edge.Azure.Monitor.MetricsScraper.ScrapeEndpoint(String endpoint, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-modules/azure-monitor/src/MetricsScraper.cs:line 72
[2021-07-13 08:15:47.044 INF] Scraping finished, received 0 metrics from endpoint http://edgeAgent:9600/metrics
[2021-07-13 08:15:47.044 INF] Scraping finished, received 0 metrics from endpoint http://edgeHub:9600/metrics
Successfully created self-signed certificate  for agentGuid : {a7673fc2-ee28-479e-b09c-355bcd1e685c} and workspace: 5967dea8-5d7d-4a56-a658-d56faf3f4956
OMS endpoint Url : https://5967dea8-5d7d-4a56-a658-d56faf3f4956.oms.opinsights.azure.com/AgentService.svc/AgentTopologyRequest
sent registration request
waiting response for registration request : OK
registration request processed
Response result status code : OK
Return Result: <?xml version="1.0" encoding="utf-8"?><AgentTopologyResponse queryInterval="PT1H" xmlns="http://schemas.microsoft.com/WorkloadMonitoring/HealthServiceProtocol/2014/09/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><ManagementGroups><ManagementGroup><Name>AOI-5967dea8-5d7d-4a56-a658-d56faf3f4956</Name><ManagementGroupId>34608ec1-1275-7d35-a782-b4484364d405</ManagementGroupId><HealthServiceId>a7673fc2-ee28-479e-b09c-355bcd1e685c</HealthServiceId><StateSynchronizationEndpoint isAuthenticated="true" queryInterval="PT5M"><Endpoint>https://5967dea8-5d7d-4a56-a658-d56faf3f4956.oms.opinsights.azure.com/ConfigurationService.Svc/GetCurrentConfigurationCookie</Endpoint><ResponseSelector>/GetConfigurationCookieResponse/ConfigurationCookie</ResponseSelector><RequestHeaders><Header>DataServiceVersion: 1.0;SCHsvc</Header><Header>MaxDataServiceVersion: 2.0;SCHsvc</Header></RequestHeaders></StateSynchronizationEndpoint><ConfigurationRequestEndpoint>https://5967dea8-5d7d-4a56-a658-d56faf3f4956.oms.opinsights.azure.com/ConfigurationService.Svc/GetConfigurationFile</ConfigurationRequestEndpoint><ManagementPackRequestEndpoint>https://5967dea8-5d7d-4a56-a658-d56faf3f4956.oms.opinsights.azure.com/ConfigurationService.Svc/GetManagementPack</ManagementPackRequestEndpoint></ManagementGroup></ManagementGroups><CertificateUpdateEndpoint updateCertificate="false">https://5967dea8-5d7d-4a56-a658-d56faf3f4956.oms.opinsights.azure.com/ConfigurationService.Svc/RenewCertificate</CertificateUpdateEndpoint></AgentTopologyResponse>
StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: private
  Server: Microsoft-IIS/10.0
  X-AspNet-Version: 4.0.30319
  X-Powered-By: ASP.NET
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  X-Content-Type-Options: nosniff
  Date: Tue, 13 Jul 2021 08:15:50 GMT
  Content-Type: application/xml; charset=utf-8
  Content-Length: 1574
}
[2021-07-13 08:15:51.789 DBG] 200 OK 
[2021-07-13 08:15:51.806 INF] Successfully sent 0 metrics to fixed set table
[2021-07-13 08:15:51.809 INF] Successfully completed periodic operation Scrape and Upload Metrics
[2021-07-13 08:20:51.811 INF] Starting periodic operation Scrape and Upload Metrics...
[2021-07-13 08:20:51.812 INF] Scraping endpoint http://edgeHub:9600/metrics
[2021-07-13 08:20:51.815 INF] Scraping endpoint http://edgeAgent:9600/metrics
[2021-07-13 08:20:51.891 ERR] Error scraping endpoint http://edgeAgent:9600/metrics - System.Net.Http.HttpRequestException: Name does not resolve
 ---> System.Net.Sockets.SocketException (0xFFFDFFFF): Name does not resolve
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.Devices.Edge.Azure.Monitor.MetricsScraper.ScrapeEndpoint(String endpoint, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-modules/azure-monitor/src/MetricsScraper.cs:line 72
[2021-07-13 08:20:51.891 INF] Scraping finished, received 0 metrics from endpoint http://edgeAgent:9600/metrics
[2021-07-13 08:20:56.838 ERR] Error scraping endpoint http://edgeHub:9600/metrics - System.Net.Http.HttpRequestException: Name does not resolve
 ---> System.Net.Sockets.SocketException (0xFFFDFFFF): Name does not resolve
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.Devices.Edge.Azure.Monitor.MetricsScraper.ScrapeEndpoint(String endpoint, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-modules/azure-monitor/src/MetricsScraper.cs:line 72
[2021-07-13 08:20:56.839 INF] Scraping finished, received 0 metrics from endpoint http://edgeHub:9600/metrics
[2021-07-13 08:20:57.846 DBG] 200 OK 
[2021-07-13 08:20:57.848 INF] Successfully sent 0 metrics to fixed set table
[2021-07-13 08:20:57.849 INF] Successfully completed periodic operation Scrape and Upload Metrics
[2021-07-13 08:25:57.848 INF] Starting periodic operation Scrape and Upload Metrics...
[2021-07-13 08:25:57.849 INF] Scraping endpoint http://edgeHub:9600/metrics
[2021-07-13 08:25:57.851 INF] Scraping endpoint http://edgeAgent:9600/metrics
[2021-07-13 08:25:57.922 ERR] Error scraping endpoint http://edgeHub:9600/metrics - System.Net.Http.HttpRequestException: Name does not resolve
 ---> System.Net.Sockets.SocketException (0xFFFDFFFF): Name does not resolve
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.Devices.Edge.Azure.Monitor.MetricsScraper.ScrapeEndpoint(String endpoint, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-modules/azure-monitor/src/MetricsScraper.cs:line 72
[2021-07-13 08:25:57.926 INF] Scraping finished, received 0 metrics from endpoint http://edgeHub:9600/metrics
[2021-07-13 08:26:02.866 ERR] Error scraping endpoint http://edgeAgent:9600/metrics - System.Net.Http.HttpRequestException: Name does not resolve
 ---> System.Net.Sockets.SocketException (0xFFFDFFFF): Name does not resolve
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.Azure.Devices.Edge.Azure.Monitor.MetricsScraper.ScrapeEndpoint(String endpoint, CancellationToken cancellationToken) in /home/vsts/work/1/s/edge-modules/azure-monitor/src/MetricsScraper.cs:line 72
[2021-07-13 08:26:02.867 INF] Scraping finished, received 0 metrics from endpoint http://edgeAgent:9600/metrics
[2021-07-13 08:26:03.628 DBG] 200 OK 
[2021-07-13 08:26:03.629 INF] Successfully sent 0 metrics to fixed set table
[2021-07-13 08:26:03.629 INF] Successfully completed periodic operation Scrape and Upload Metrics

**Comment by veyalla** [comment]
Hmm, the only thing I can think of is the modules are not on the same docker network, but I presume you checked that since it was mentioned in the monitoring troubleshooting steps? Can you share the output of:

sudo docker inspect edgeHub
sudo docker inspect edgeAgent
sudo docker inspect IoTEdgeMetricsCollector

**Comment by Ilheu** [comment]
**sudo docker inspect edgeAgent**
[
    {
        "Id": "b5f0fa4c25b886f89eb1bec457d6c7cd4f754c0a2ba23a071f18eee2f287c6af",
        "Created": "2021-07-13T08:14:34.148309182Z",
        "Path": "/bin/sh",
        "Args": [
            "-c",
            "exec /app/agentStart.sh"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 30853,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2021-07-13T09:01:05.123864453Z",
            "FinishedAt": "2021-07-13T09:01:02.376818405Z"
        },
        "Image": "sha256:97834f10b7b0f7903d28c9fb32ceae672a8a66ba1ea868739fc9818e184557c2",
        "ResolvConfPath": "/var/lib/docker/containers/b5f0fa4c25b886f89eb1bec457d6c7cd4f754c0a2ba23a071f18eee2f287c6af/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/b5f0fa4c25b886f89eb1bec457d6c7cd4f754c0a2ba23a071f18eee2f287c6af/hostname",
        "HostsPath": "/var/lib/docker/containers/b5f0fa4c25b886f89eb1bec457d6c7cd4f754c0a2ba23a071f18eee2f287c6af/hosts",
        "LogPath": "/var/lib/docker/containers/b5f0fa4c25b886f89eb1bec457d6c7cd4f754c0a2ba23a071f18eee2f287c6af/b5f0fa4c25b886f89eb1bec457d6c7cd4f754c0a2ba23a071f18eee2f287c6af-json.log",
        "Name": "/edgeAgent",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/var/run/iotedge/workload.sock:/var/run/iotedge/workload.sock",
                "/var/run/iotedge/mgmt.sock:/var/run/iotedge/mgmt.sock"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "9600/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "9601"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Capabilities": null,
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/5cf41a2b8f6689ad4aca2e57a4e727ba6e195d1329869778704163b2aa4dd9c9-init/diff:/var/lib/docker/overlay2/e8e069f7490793860adb679aaae4ea8abfb4ceedfbbeb74564aa6946b1b18f2f/diff:/var/lib/docker/overlay2/c75ab2d65cbfefa65c92e9bdf57abb85891ec57a6f144d6f67fd8950cc6b7de8/diff:/var/lib/docker/overlay2/f73590bde2650918f996df2861a321e76d315849aab75e9be40e7094f6224529/diff:/var/lib/docker/overlay2/74672c2b67d1858718d7532e861343108dd9b6207c4b3b3e98965184f8265e27/diff:/var/lib/docker/overlay2/dab8fbc980b0caa29b6b3a16e0a0ad00d4bcb6bcfe646a64d4662b313cf2e12f/diff:/var/lib/docker/overlay2/c959676af603669f731ff93b9c43f58a24f67597012ac6dbe8021fb721829fb8/diff:/var/lib/docker/overlay2/a958a9f722f80e02076bf6af39b2eddb0a293369c08aa5f95b02bbc00d36546d/diff:/var/lib/docker/overlay2/d5e19eb419e8d566afbbdca17f1458e4323fc48f80aac0fbc611b6fd4cbbb1fc/diff:/var/lib/docker/overlay2/74140cdf0c1c4f616531c1a0b9a9f846a48d5b3781fc03bb047e02a8f81cbf38/diff",
                "MergedDir": "/var/lib/docker/overlay2/5cf41a2b8f6689ad4aca2e57a4e727ba6e195d1329869778704163b2aa4dd9c9/merged",
                "UpperDir": "/var/lib/docker/overlay2/5cf41a2b8f6689ad4aca2e57a4e727ba6e195d1329869778704163b2aa4dd9c9/diff",
                "WorkDir": "/var/lib/docker/overlay2/5cf41a2b8f6689ad4aca2e57a4e727ba6e195d1329869778704163b2aa4dd9c9/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/var/run/iotedge/workload.sock",
                "Destination": "/var/run/iotedge/workload.sock",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/var/run/iotedge/mgmt.sock",
                "Destination": "/var/run/iotedge/mgmt.sock",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "b5f0fa4c25b8",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "EdgeDeviceHostName=muvu-tech",
                "IOTEDGE_APIVERSION=2020-07-07",
                "IOTEDGE_AUTHSCHEME=sasToken",
                "IOTEDGE_DEVICEID=edgePrototipo",
                "IOTEDGE_IOTHUBHOSTNAME=MuvuIoT.azure-devices.net",
                "IOTEDGE_MANAGEMENTURI=unix:///var/run/iotedge/mgmt.sock",
                "IOTEDGE_MODULEGENERATIONID=637595253242541829",
                "IOTEDGE_MODULEID=$edgeAgent",
                "IOTEDGE_WORKLOADURI=unix:///var/run/iotedge/workload.sock",
                "Mode=iotedged",
                "NetworkId=azure-iot-edge",
                "RuntimeLogLevel=Information",
                "httpSettings__enabled=true",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "ASPNETCORE_URLS=http://+:80",
                "DOTNET_RUNNING_IN_CONTAINER=true",
                "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=true",
                "EDGEAGENTUSER_ID=1000"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "exec /app/agentStart.sh"
            ],
            "Image": "mcr.microsoft.com/azureiotedge-agent:1.1",
            "Volumes": null,
            "WorkingDir": "/app",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "net.azure-devices.edge.create-options": "{\"HostConfig\":{\"PortBindings\":{\"9600/tcp\":[{\"HostPort\":\"9601\"}]}}}",
                "net.azure-devices.edge.env": "{\"httpSettings__enabled\":{\"value\":\"true\"}}",
                "net.azure-devices.edge.original-image": "mcr.microsoft.com/azureiotedge-agent:1.1",
                "net.azure-devices.edge.owner": "Microsoft.Azure.Devices.Edge.Agent"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "746c8ddf0f22c7af96f537f2bc1650b98bdc78bdf3a1cfe8ea6c4ad601dcef46",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/746c8ddf0f22",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "azure-iot-edge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "b5f0fa4c25b8"
                    ],
                    "NetworkID": "bc0bd72b2d3ec7f4140d3ee30780b9223df2a4f2f2554c69cbfe8532487195ae",
                    "EndpointID": "476ef7bf8425eac03e30c6ba4be505099ccd21409ccb8f19c4a0387a943f142f",
                    "Gateway": "172.18.0.1",
                    "IPAddress": "172.18.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:12:00:02",
                    "DriverOpts": null
                }
            }
        }
    }
]
**------------------------------------------------------------------------------------------------------------------------------**






**------------------------------------------------------------------------------------------------------------------------------**

**Comment by veyalla** [comment]
Thanks for the info. Nothing obvious jumps out. Few suggestions:

* Update to the latest Moby version, using these [instructions](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge?view=iotedge-2020-11#install-a-container-engine).
* Remove these createOptions from edgeAgent: `"net.azure-devices.edge.create-options": "{"HostConfig":{"PortBindings":{"9600/tcp":[{"HostPort":"9601"}]}}}",`. They are not needed for metrics-collector to work.
* Update to latest metrics-collector using an explicit tag: `mcr.microsoft.com/azureiotedge-metrics-collector:1.0.1`

If this still doesn't help, we'll need to take deeper look, so please open an [Azure support ticket](https://azure.microsoft.com/en-us/support/create-ticket/). You can reference this GitHub issue in the support ticket.

FYI @and-rewsmith

**Comment by Ilheu** [comment]
It is working! 
I uninstalled and installed again the edge and the moby. So I suppose it was the Moby version.
@veyalla thank you very much for your patience and quick reply!
